### ISSUE TITLE
Inflation in interaction p-values

### ISSUE THREAD

**Issue by came203** [rgcgithub/regenie#263]
Hi Joelle,

I was testing out some of the new interaction analyses and found something that I can't really explain:

I'm looking at a quantitative imaging phenotype in UKB Europeans (N=33,000) and want to see interaction of the imputed UKB SNPs with a rare environmental binary exposure (present only in 75/33,000 IDs) that I put in the cat_covar list in step2. 
I understand that the power to detect something here is miniscule, especially for low-frequency variants, but I nevertheless ran the analysis in PLINK2 and now in regenie v3.
I'm only looking at the term "ADD-INT_SNPxVAR=1" which corresponds to the PLINK term "ADDxVAR"

It seems that for **MAF > 10%**, everything works out nicely, the resulting p-values from PLINK2 and REGENIE seem to correlate quite well (Spearman's r=0.9) with some outliers as expected. In general, p-values in REGENIE are more conservative than PLINK.
![NOTCH3_WMH_MAF01](https://user-images.githubusercontent.com/48828321/159009686-fe676dbc-a2c1-4efe-b0b2-147ca62816e4.jpeg)

QQ-Plot REGENIE - seem to be fine and not too inflated
![grafik](https://user-images.githubusercontent.com/48828321/159024096-8938cb69-cd6e-4997-9c2b-78277b98ac2c.png)

QQ-PLOT PLINK - a bit inflated
![grafik](https://user-images.githubusercontent.com/48828321/159027892-4a2a084b-e0d5-4c84-bfd1-fa0c83e86006.png)

**However, once I go to MAF < 10%, things start getting weird**. P-values are heavily inflated in the REGENIE results, up to -log(P) of 150

![WMH_MAFlower01](https://user-images.githubusercontent.com/48828321/159010161-8e218cd1-efd5-43b2-8c51-48e28c5f951c.jpeg)

QQ-Plot REGENIE: Here you can see the extreme inflation
![grafik](https://user-images.githubusercontent.com/48828321/159018178-dc6263a9-7a59-4b2a-bc97-e31f142a4abc.png)

In PLINK, the p-values are still quite well-behaved although a bit inflated
QQ-Plot PLINK
![QQ_PLINK](https://user-images.githubusercontent.com/48828321/159014525-16f1b9b6-e243-42e7-a368-6c6d867ea4b5.jpeg)

I'm trying to understand what happens here, I doesn't seem to happen with the binary outcome phenotypes I tested (same environmental exposure). I haven't tested a quantitative outcome phenotype that's available in the whole dataset yet.

Rainer

**Comment by joellembatchou** [comment]
Hi Rainer,

We did not consider the scenario where the risk factor was this rare so the current approach of using robust SE for variants with MAC>1000 (by default) may not be optimal here; similarly to how rare variants tested in GxE with robust SE can give inflation, the exposure being very rare can make it worse even if the variant tested is not so rare. 

The heteroskedastic linear model (HLM) model does not suffer from this limitation of the robust SE so can you re-do the run but this time enforcing that HLM model be used for all variants (ie set high MAC threshold for HLM;eg `--rare-mac 1000000000`). How do the results look when you only use HLM strategy for GxE test?

**Comment by Shicheng-Guo** [comment]
Agree, robust SE will be very sensitive to rare variants and the association will be significant inflated. I am wondering what's the best cut-off for MAC and frequency of exposure? Hope to see some benchmarking test. Thanks. 
Shicheng

**Comment by came203** [comment]
Dear all,

I re-ran the same analysis using the` --rare-mac 1000000000`. It seems like the p-values are now heavily deflated:

![grafik](https://user-images.githubusercontent.com/48828321/159457485-3e07294d-ae78-4393-90a8-567229c1b92a.png)


Note that these are all variants regardless of MAF. 

Here are the variants <10% MAF that caused the problems before:
![grafik](https://user-images.githubusercontent.com/48828321/159467184-e48942aa-8e0d-43c8-9b4f-1bbb31ac7a83.png)

Again, deflation of p-values happens here.

Best wishes
Rainer
